### DISCUSSION TITLE
Precise Wall function

### DISCUSSION THREAD

**Discussion by mbrady4597** [OrcaSlicer/OrcaSlicer#831] [General]
Being a main feature of the Orcaslicer fork, why is this feature not explained in detail anywhere? The brief explanation in the release notes is fairly short and vague. It would be great to have a bit more info on how it works to have confidence that it isn't causing any unwanted behavior or conflict with other settings.

**Comment by EvaFPV** [comment]
I'd like to know more too. Having it on seems to be better than off, some holes I've needed interference fits from are slightly oversized with it off, and much closer to tolerance with it on. It looks like in Arachne mode it's slightly increasing the line width of outer perimeters thus overriding the outer perimeter wall thickness setting. Haven't tried printing with it on and off in Classic mode to see if there's any effect there. Doesn't look like there's a visible change in the sliced preview though. 

"Improve shell precision by adjusting outer wall spacing. This also improves layer consistency" is an incredibly vague tool-tip, especially for a setting that actually has a significant impact on printing tolerances.

**Comment by julie777** [comment]
Me too. I created an issue requesting that documentation keep up with features and settings.

**Comment by capposteve** [comment]
@SoftFever could you please let us folk know what this setting is meant to change etc. What are its main benefits and uses? I understand its experimental however still interested to know. Ive studied how it works on layers and i noticed some odd behavior but would like to know if this is normal. Thanks

**Comment by scottmudge** [comment]
Looking at the code, what it appears to do is slightly shift the position of the outer wall. By default, extruded lines use the "spacing" between lines to set their position. The "precise wall" mode uses the actual "width" of the line instead.

Extruded paths have a Flow property/object assigned to them. This describes the amount of material used per unit distance. This Flow object contains a `width` property, and a `spacing` property.

The `width` is the actual calculated with of the line. E.g., if you extruded a single line and measured its width with calipers.

However, if you simply extruded multiple walls next to each other, and spaced them apart using their theoretical width, they'd bunch up and create bulges as the rounded edges of the lines pushed up against eachother.

The developers of PrusaSlicer got around this by adding a small gap between the lines. The gap + the theoretical width yields the "spacing" attribute.

So the idea here is that using the `spacing` property (theoretical width + gap between adjacent lines/paths) for the outer wall means the outer wall isn't *exactly* where it should be. I'm not sure if this actually true or not (my experience is that the gap doesn't matter), but the "Precise Wall" mode ignores this gap, and uses the theoretical width of the line for the outer wall instead.

It's not really doing anything else. From what I can tell, it's effectively *slightly* shrinking the outside wall by moving it in just a bit.

In practice, the difference may be negligible. Models printed with "Precise mode" off on a properly calibrated printer show (for me) accuracy within +/- 0.01mm, so I'm not really sure if the potential for bunched up ooze between the outer wall and the adjacent wall is worth it.

**Comment by SoftFever** [comment]
Sure, I can give a brief explanation of what I did for the precise wall feature.

First, it's important to understand some basic concepts like flow, extrusion width, and space. Slic3r has an excellent document that covers these topics in detail. You can refer to this article: [link to article](https://manual.slic3r.org/advanced/flow-math).

Now, let's dive into the specifics. Slic3r and its forks, such as PrusaSlicer, SuperSlicer, BambuSlicer, and OrcaSlicer, assume that the extrusion path has an oval shape, which accounts for the overlaps. For example, if we set the wall width to 0.4mm and the layer height to 0.2mm, the combined thickness of two walls laid side by side is 0.714mm instead of 0.8mm due to the overlapping.

This approach enhances the strength of 3D-printed parts. However, it does have some side effects. For instance, when the inner-outer wall order is used, the outer wall can be pushed outside, leading to potential size inaccuracy and more layer inconsistency.

Secondly, it's important to note that this is Slic3r's approach to handling flow. Other slicers like Cura assume that the extrusion path is rectangular and do not have overlapping.

Personally, I prefer Slic3r's approach overall. To address the downsides I mentioned earlier, I introduced the precise wall feature. When the precise wall feature is enabled in OrcaSlicer, the overlap between the outer wall and its neighboring inner wall is set to 0. This means that the overall strength of the printed part is not affected, but the size and layer consistency are improved.

I hope this explanation clarifies the purpose and benefits of the precise wall feature.

**Reply by dbaarda** [reply]
IMHO SlicEr's approach is a bit of an unnecessary complication because after all the modelling it falls back to using an overlap-factor of 1.0, which means it assumes the adjacent line edges get squished into a rectangular profile and the line-spacing is exactly the same as the line-width in the rectangular flow model. The one place it's kind of useful is on the outer-edges which don't get squished flat, but even then it might not always be useful. See my thoughts and maths for this at; https://github.com/SoftFever/OrcaSlicer/issues/8313#issuecomment-3082244171

**Comment by mbrady4597** [comment]
Thanks, that clears it up!

**Comment by dpjrodrigues** [comment]
When printing the calibration cube (04 nozzle, no infill, no top layers and 2 walls (0.4 each), I've realised that when precise wall is on the slicer creates a inner wall with 0.4 and a outer-wall with 0.53+... this results in a very clean cube but way off in terms of expected wall width, i.e. 0.93 instead of 0.8 (without the precise wall I get 0.76 instead of 0.8... which is kind of expected due to the overlap... but not correct also)

Shouldn't this feature eliminate just the overlap? in my case it seems that it is increasing the width of the outer layer. Can send more details if necessary

**Comment by sleewok** [comment]
I have noticed that there appears to be an issue with Arachne and precise wall. For example, you can see in the first image here how the lines cross the perimeter of the object. This is with Arachne and precise wall enabled.

![image](https://github.com/SoftFever/OrcaSlicer/assets/304940/4bd94fb5-e457-481c-9c8c-f7c5b16ad0ff)

When I disable precise wall this is the result:

![image](https://github.com/SoftFever/OrcaSlicer/assets/304940/fad4fe0a-54db-472f-9e63-ed59e8c7dd27)

**Reply by SoftFever** [reply]
You are right.
This didn't work as expected when with Arachne engine.

**Reply by SoftFever** [reply]
@sleewok this should be fixed #2966

**Comment by xenon2008** [comment]
where could i activate this "precise wall" function in Orca Slicer?

**Comment by evranch** [comment]
Could we get a filament-specific Setting Override for Precise Wall like there are for retraction?
I've been happy with Precise Wall with solid filaments, but it caused delamination of the outer wall when I tried it with TPU. Not enough overlap.

![IMG_20231221_092107](https://github.com/SoftFever/OrcaSlicer/assets/8278477/fd5549c4-d334-4ca2-ab10-901ba7f8f584)

@xenon2008 The setting can be found under Quality -> Precision

**Comment by constant-flow** [comment]
I asked myself and the internet: [Wouldn't a higher flowrate for 2nd outer wall increase the print quality?](https://3dprinting.stackexchange.com/questions/15762/wouldnt-a-higher-flowrate-for-2nd-outer-wall-increase-the-print-quality)

I feel like this **Precise Wall** setting is tackling the same issue, with a slight difference, but maybe my thoughts spark some ideas and may even help with artifacts like delamination shown above.

(Disclaimer: never used slic3r nor precise wall ... just stumbled over it and remembered me)

**Reply by mbrady4597** [reply]
Don’t have a stack exchange account so I’ll reply here. Unfortunately the theoretical drawing there doesn’t exactly match reality and depends on a number of factors. The reality is that any inner wall line pushed up against the most outer wall will cause dimensional issues. The squish will deform and push the outer line outwards instead of perfectly filling the gaps. This effect is dependent on material properties, print temperature, cooling, and flow rate. That is why the precise wall feature reduces flow for the outer wall- to ensure it isn’t pushed outward by the inner walls. 

The idea you have can already be accomplished in orca slicer though. Just set wall order to print accordingly and use a slightly higher infill for inner walls. The excess material tends to accumulate and squish outward over a number of walls because the nozzle prevents it from pushing upward, so this would lead to the thicker “2nd outer wall” that you suggest. 

The best way to compensate for dimensional inaccuracies for prints that need strong bonding is compensating in the modeling. This requires testing to accomplish.

**Comment by JannetGen** [comment]
I’d try changing one setting at a time and run a small test print after each change. It makes it much easier to figure out which setting is actually affecting the result. Once you get a good combination, save it as a profile so you don’t have to repeat the tuning later.
